x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -28,22 +30,30 @@ import {
   EmbeddableInput,
   EmbeddableOutput,
   IContainer,
+  SavedObjectEmbeddableInput,
+  ReferenceOrValueEmbeddable,
 } from '../../../../../../src/plugins/embeddable/public';
 import { DOC_TYPE, Document, injectFilterReferences } from '../../persistence';
 import { ExpressionWrapper } from './expression_wrapper';
 import { UiActionsStart } from '../../../../../../src/plugins/ui_actions/public';
 import { isLensBrushEvent, isLensFilterEvent } from '../../types';
 
-export interface LensEmbeddableConfiguration {
-  expression: string | null;
-  savedVis: Document;
-  editUrl: string;
-  editPath: string;
-  editable: boolean;
-  indexPatterns?: IIndexPattern[];
-}
+import { IndexPatternsContract } from '../../../../../../src/plugins/data/public';
+import { getEditPath } from '../../../common';
+import { IBasePath, SavedObject } from '../../../../../../src/core/public';
+import { AttributeService } from '../../../../../../src/plugins/dashboard/public';
+
+export type LensSavedObjectAttributes = Omit<Document, 'id' | 'type'>;
 
-export interface LensEmbeddableInput extends EmbeddableInput {
+export type LensByValueInput = {
+  attributes: LensSavedObjectAttributes;
+  references: SavedObject<LensSavedObjectAttributes>['references'];
+} & LensInheritedInput;
+
+export type LensByReferenceInput = SavedObjectEmbeddableInput & LensInheritedInput;
+export type LensEmbeddableInput = LensByValueInput | LensByReferenceInput;
+
+export interface LensInheritedInput extends EmbeddableInput {
```

Comment:
Some of these types are outdated. In particular, `LensEmbeddableInput` is easy to simplify because `EmbeddableInput` already contains timeRange, query, filters.

```suggestion
export type LensSavedObjectAttributes = Omit<Document, 'type'>;

export type LensByValueInput = {
  attributes: LensSavedObjectAttributes;
  references: SavedObject<LensSavedObjectAttributes>['references'];
} & EmbeddableInput;

export type LensByReferenceInput = SavedObjectEmbeddableInput & EmbeddableInput;

export type LensEmbeddableInput = LensByValueInput | LensByReferenceInput;
```

Reply:
Fixed these up in the next version. It's much cleaner this way, thank you!